lib/generators/rspec/scaffold/scaffold_generator.rb

```diff
@@ -7,17 +7,32 @@ module Generators
     class ScaffoldGenerator < Base
       include ::Rails::Generators::ResourceHelpers
       source_paths << File.expand_path('../helper/templates', __dir__)
-      argument :attributes, :type => :array, :default => [], :banner => "field:type field:type"
 
-      class_option :orm, :desc => "ORM used to generate the controller"
-      class_option :template_engine, :desc => "Template engine to generate view files"
-      class_option :singleton, :type => :boolean, :desc => "Supply to create a singleton controller"
-      class_option :api, :type => :boolean, :desc => "Skip specs unnecessary for API-only apps"
-
-      class_option :controller_specs, :type => :boolean, :default => true,  :desc => "Generate controller specs"
-      class_option :view_specs,       :type => :boolean, :default => true,  :desc => "Generate view specs"
-      class_option :helper_specs,     :type => :boolean, :default => true,  :desc => "Generate helper specs"
-      class_option :routing_specs,    :type => :boolean, :default => true,  :desc => "Generate routing specs"
+      if Rails.version.to_f >= 4.2
+        argument :attributes, type: :array, default: [], banner: "field:type field:type"
+
+        class_option :orm, desc: "ORM used to generate the controller"
+        class_option :template_engine, desc: "Template engine to generate view files"
+        class_option :singleton, type: :boolean, desc: "Supply to create a singleton controller"
+        class_option :api, type: :boolean, desc: "Skip specs unnecessary for API-only apps"
+
+        class_option :controller_specs, type: :boolean, default: true,  desc: "Generate controller specs"
+        class_option :view_specs,       type: :boolean, default: true,  desc: "Generate view specs"
+        class_option :helper_specs,     type: :boolean, default: true,  desc: "Generate helper specs"
+        class_option :routing_specs,    type: :boolean, default: true,  desc: "Generate routing specs"
+      else
```

Comment:
I believe this branch can be dropped altogether.

Reply:
I'm wondering to see that when change has tag, the CI builds fails due to unknown hash subject.
I'm trying hard but could not understand why test is being failed.

Reply:
Can you please point at a build with the failure you mention?

I can only see [RuboCop complaining](https://travis-ci.org/rspec/rspec-rails/jobs/615437211) at `Style/HashSyntax: Use hash rockets syntax`, feel free to update `.rubocop.yml`'s rule to prefer the new hash syntax and this offence will go away.